File: app/src/main/java/tomaszkopacz/meetbam/dialogs/ChooseDeviceDialog.java
package tomaszkopacz.meetbam.dialogs;

import android.app.Dialog;
import android.bluetooth.BluetoothDevice;
import android.content.Context;
import android.widget.RadioButton;
import android.widget.RadioGroup;

import java.util.ArrayList;
import java.util.List;

import butterknife.BindView;
import butterknife.ButterKnife;
import tomaszkopacz.meetbam.R;
import tomaszkopacz.meetbam.presenters.MainActivityPresenter;

/**
 * Created by tomas on 27.05.2018.
 */

public class ChooseDeviceDialog extends Dialog {

    private Context context;
    private MainActivityPresenter presenter;

    private List<BluetoothDevice> devices = new ArrayList<>();

    @BindView(R.id.devices_group)
    RadioGroup group;

    public ChooseDeviceDialog(Context context, MainActivityPresenter presenter) {
        super(context);

        this.context = context;
        this.presenter = presenter;
        setContentView(R.layout.dialog_choose_device);
        ButterKnife.bind(this);

        setTitle("Choose device:");
        group.setOnCheckedChangeListener(listener);

    }

    public void setDevices(List<BluetoothDevice> devices){

        this.devices.clear();
        group.removeAllViews();

        this.devices = devices;

        for (int i = 0; i < devices.size(); i++){
            RadioButton btn = new RadioButton(context);
            btn.setText(devices.get(i).getName());
            btn.setId(i);
            group.addView(btn);
        }
    }

    private RadioGroup.OnCheckedChangeListener listener = new RadioGroup.OnCheckedChangeListener() {
        @Override
        public void onCheckedChanged(RadioGroup radioGroup, int i) {

            int selectedItem = radioGroup.getCheckedRadioButtonId();
            presenter.deviceSelected(devices.get(selectedItem).getAddress());
        }
    };
}

File: app/src/main/java/tomaszkopacz/meetbam/dialogs/ChooseDeviceDialog.kt
package tomaszkopacz.meetbam.dialogs

import android.app.Dialog
import android.content.Context
import tomaszkopacz.meetbam.presenter.MainPostsFragmentPresenter

class ChooseDeviceDialog(context: Context, presenter: MainPostsFragmentPresenter) : Dialog(context) {
//
//    private var devices: MutableList<BluetoothDevice> = ArrayList()
//    val group: RadioGroup by bindView(R.id.devices_group)
//
//    private val listener = RadioGroup.OnCheckedChangeListener {
//        radioGroup, _ ->
//            val selectedItem = radioGroup.checkedRadioButtonId
//            //presenter.deviceSelected(devices[selectedItem])
//    }
//
//    init {
//        setContentView(R.layout.dialog_choose_device)
//
//        setTitle("Choose device:")
//        group.setOnCheckedChangeListener(listener)
//
//    }
//
//    fun setDevices(devices: MutableList<BluetoothDevice>) {
//
//        this.devices.clear()
//        group.removeAllViews()
//
//        this.devices = devices
//
//        for (i in devices.indices) {
//            val btn = RadioButton(context)
//            btn.text = devices[i].name
//            btn.id = i
//            group.addView(btn)
//        }
//    }
}
